[ci] `sqlite3.OperationalError: no such table`

### What happened?
Both the `//examples/hardware_sim:py/hardware_sim_py_test` and  `//bindings/pydrake/common:py/text_logging_test` tests fail here:

https://drake-jenkins.csail.mit.edu/job/linux-noble-gcc-bazel-nightly-coverage/317/console


### Version

_No response_

### What operating system are you using?

Ubuntu 24.04 (Noble)

### What installation option are you using?

_No response_

### Relevant log output

```shell
[6:14:05 AM]  FAIL: //examples/hardware_sim:py/hardware_sim_py_test (shard 1 of 4) (Exit 1) (see /media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/testlogs/examples/hardware_sim/py/hardware_sim_py_test/shard_1_of_4/test.log)
[6:14:05 AM]  INFO: From Testing //examples/hardware_sim:py/hardware_sim_py_test (shard 1 of 4):
[6:14:05 AM]  ==================== Test output for //examples/hardware_sim:py/hardware_sim_py_test (shard 1 of 4):
[6:14:05 AM]  info: Shard 1 / 4 has these tests:
[6:14:05 AM]  info: - test_Defaults (hardware_sim_py_test.HardwareSimPyTest.test_Defaults)
[6:14:05 AM]  
[6:14:05 AM]  Running tests...
[6:14:05 AM]  ----------------------------------------------------------------------
[6:14:05 AM]  == Running _main/examples/hardware_sim/hardware_sim_py --scenario_file=_main/examples/hardware_sim/test/test_scenarios.yaml --scenario_name=Defaults '--scenario_text={simulation_duration: 0.0625}'
[6:14:05 AM]  INFO:drake:LCM bus 'default' created for URL memq://
[6:14:05 AM]  INFO:drake:Meshcat listening for connections at http://localhost:7000/
[6:14:05 AM]  .
[6:14:05 AM]  ----------------------------------------------------------------------
[6:14:05 AM]  Ran 1 test in 6.594s
[6:14:05 AM]  
[6:14:05 AM]  OK
[6:14:05 AM]  
[6:14:05 AM]  Generating XML reports...
[6:14:05 AM]  Traceback (most recent call last):
[6:14:05 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/examples/hardware_sim/py/hardware_sim_py_test.runfiles/_main/examples/hardware_sim/py/_hardware_sim_py_test_stage2_bootstrap.py", line 393, in _maybe_collect_coverage
[6:14:05 AM]      yield
[6:14:05 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/examples/hardware_sim/py/hardware_sim_py_test.runfiles/_main/examples/hardware_sim/py/_hardware_sim_py_test_stage2_bootstrap.py", line 531, in main
[6:14:05 AM]      _run_py_path(main_filename, args=sys.argv[1:])
[6:14:05 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/examples/hardware_sim/py/hardware_sim_py_test.runfiles/_main/examples/hardware_sim/py/_hardware_sim_py_test_stage2_bootstrap.py", line 319, in _run_py_path
[6:14:05 AM]      runpy.run_path(main_filename, run_name="__main__")
[6:14:05 AM]    File "<frozen runpy>", line 286, in run_path
[6:14:05 AM]    File "<frozen runpy>", line 98, in _run_module_code
[6:14:05 AM]    File "<frozen runpy>", line 88, in _run_code
[6:14:05 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/examples/hardware_sim/py/hardware_sim_py_test.runfiles/_main/common/test_utilities/drake_py_unittest_main.py", line 270, in <module>
[6:14:05 AM]      main()
[6:14:05 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/examples/hardware_sim/py/hardware_sim_py_test.runfiles/_main/common/test_utilities/drake_py_unittest_main.py", line 218, in main
[6:14:05 AM]      run()
[6:14:05 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/examples/hardware_sim/py/hardware_sim_py_test.runfiles/_main/common/test_utilities/drake_py_unittest_main.py", line 195, in run
[6:14:05 AM]      _unittest_main(
[6:14:05 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/examples/hardware_sim/py/hardware_sim_py_test.runfiles/_main/common/test_utilities/drake_py_unittest_main.py", line 73, in _unittest_main
[6:14:05 AM]      sys.exit(0 if result.wasSuccessful() else 1)
[6:14:05 AM]  SystemExit: 0
[6:14:05 AM]  
[6:14:05 AM]  During handling of the above exception, another exception occurred:
[6:14:05 AM]  
[6:14:05 AM]  Traceback (most recent call last):
[6:14:05 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 114, in _execute
[6:14:05 AM]      return self.con.execute(sql, parameters)    # type: ignore[arg-type]
[6:14:05 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:05 AM]  sqlite3.OperationalError: no such table: arc
[6:14:05 AM]  
[6:14:05 AM]  During handling of the above exception, another exception occurred:
[6:14:05 AM]  
[6:14:05 AM]  Traceback (most recent call last):
[6:14:05 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 119, in _execute
[6:14:05 AM]      return self.con.execute(sql, parameters)    # type: ignore[arg-type]
[6:14:05 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:05 AM]  sqlite3.OperationalError: no such table: arc
[6:14:05 AM]  
[6:14:05 AM]  The above exception was the direct cause of the following exception:
[6:14:05 AM]  
[6:14:05 AM]  Traceback (most recent call last):
[6:14:05 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/examples/hardware_sim/py/hardware_sim_py_test.runfiles/_main/examples/hardware_sim/py/_hardware_sim_py_test_stage2_bootstrap.py", line 537, in <module>
[6:14:05 AM]      main()
[6:14:05 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/examples/hardware_sim/py/hardware_sim_py_test.runfiles/_main/examples/hardware_sim/py/_hardware_sim_py_test_stage2_bootstrap.py", line 528, in main
[6:14:05 AM]      with _maybe_collect_coverage(enable=coverage_enabled):
[6:14:05 AM]    File "/usr/lib/python3.12/contextlib.py", line 158, in __exit__
[6:14:05 AM]      self.gen.throw(value)
[6:14:05 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/examples/hardware_sim/py/hardware_sim_py_test.runfiles/_main/examples/hardware_sim/py/_hardware_sim_py_test_stage2_bootstrap.py", line 398, in _maybe_collect_coverage
[6:14:05 AM]      cov.lcov_report(
[6:14:05 AM]    File "/usr/lib/python3/dist-packages/coverage/control.py", line 1277, in lcov_report
[6:14:05 AM]      return render_report(self.config.lcov_output, LcovReporter(self), morfs, self._message)
[6:14:05 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:05 AM]    File "/usr/lib/python3/dist-packages/coverage/report_core.py", line 61, in render_report
[6:14:05 AM]      ret = reporter.report(morfs, outfile=outfile)
[6:14:05 AM]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:05 AM]    File "/usr/lib/python3/dist-packages/coverage/lcovreport.py", line 50, in report
[6:14:05 AM]      self.get_lcov(fr, analysis, outfile)
[6:14:05 AM]    File "/usr/lib/python3/dist-packages/coverage/lcovreport.py", line 100, in get_lcov
[6:14:05 AM]      sorted(analysis.branch_stats().keys()),
[6:14:05 AM]             ^^^^^^^^^^^^^^^^^^^^^^^
[6:14:05 AM]    File "/usr/lib/python3/dist-packages/coverage/results.py", line 172, in branch_stats
[6:14:05 AM]      missing_arcs = self.missing_branch_arcs()
[6:14:05 AM]                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:05 AM]    File "/usr/lib/python3/dist-packages/coverage/results.py", line 143, in missing_branch_arcs
[6:14:05 AM]      missing = self.arcs_missing()
[6:14:05 AM]                ^^^^^^^^^^^^^^^^^^^
[6:14:05 AM]    File "/usr/lib/python3/dist-packages/coverage/results.py", line 103, in arcs_missing
[6:14:05 AM]      executed = self.arcs_executed()
[6:14:05 AM]                 ^^^^^^^^^^^^^^^^^^^^
[6:14:05 AM]    File "/usr/lib/python3/dist-packages/coverage/results.py", line 96, in arcs_executed
[6:14:05 AM]      executed = self.data.arcs(self.filename) or []
[6:14:05 AM]                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:05 AM]    File "/usr/lib/python3/dist-packages/coverage/sqldata.py", line 1006, in arcs
[6:14:05 AM]      with con.execute(query, data) as cur:
[6:14:05 AM]    File "/usr/lib/python3.12/contextlib.py", line 137, in __enter__
[6:14:05 AM]      return next(self.gen)
[6:14:05 AM]             ^^^^^^^^^^^^^^
[6:14:05 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 149, in execute
[6:14:05 AM]      cur = self._execute(sql, parameters)
[6:14:05 AM]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:05 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 137, in _execute
[6:14:05 AM]      raise DataError(f"Couldn't use data file {self.filename!r}: {msg}") from exc
[6:14:05 AM]  coverage.exceptions.DataError: Couldn't use data file '/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/examples/hardware_sim/py/hardware_sim_py_test.runfiles/_main/.coverage': no such table: arc
```

also:
```shell
FAIL: //bindings/pydrake/common:py/text_logging_test (shard 2 of 16) (Exit 1) (see /media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/testlogs/bindings/pydrake/common/py/text_logging_test/shard_2_of_16/test.log)
[6:14:54 AM]  INFO: From Testing //bindings/pydrake/common:py/text_logging_test (shard 2 of 16):
[6:14:54 AM]  ==================== Test output for //bindings/pydrake/common:py/text_logging_test (shard 2 of 16):
[6:14:54 AM]  info: Shard 2 / 16 has these tests:
[6:14:54 AM]  info: - test_example_false_false_0_0 (text_logging_test.TestTextLoggingExample.test_example_false_false_0_0)
[6:14:54 AM]  info: - test_example_false_false_1_0 (text_logging_test.TestTextLoggingExample.test_example_false_false_1_0)
[6:14:54 AM]  info: - test_example_false_false_30_0 (text_logging_test.TestTextLoggingExample.test_example_false_false_30_0)
[6:14:54 AM]  info: - test_example_false_false_50_0 (text_logging_test.TestTextLoggingExample.test_example_false_false_50_0)
[6:14:54 AM]  info: - test_example_false_true_0_0 (text_logging_test.TestTextLoggingExample.test_example_false_true_0_0)
[6:14:54 AM]  info: - test_example_false_true_1_0 (text_logging_test.TestTextLoggingExample.test_example_false_true_1_0)
[6:14:54 AM]  info: - test_example_false_true_30_0 (text_logging_test.TestTextLoggingExample.test_example_false_true_30_0)
[6:14:54 AM]  info: - test_example_false_true_50_0 (text_logging_test.TestTextLoggingExample.test_example_false_true_50_0)
[6:14:54 AM]  info: - test_example_true_false_0_0 (text_logging_test.TestTextLoggingExample.test_example_true_false_0_0)
[6:14:54 AM]  info: - test_example_true_false_1_0 (text_logging_test.TestTextLoggingExample.test_example_true_false_1_0)
[6:14:54 AM]  info: - test_example_true_false_30_0 (text_logging_test.TestTextLoggingExample.test_example_true_false_30_0)
[6:14:54 AM]  info: - test_example_true_false_50_0 (text_logging_test.TestTextLoggingExample.test_example_true_false_50_0)
[6:14:54 AM]  info: - test_example_true_true_0_0 (text_logging_test.TestTextLoggingExample.test_example_true_true_0_0)
[6:14:54 AM]  info: - test_example_true_true_1_0 (text_logging_test.TestTextLoggingExample.test_example_true_true_1_0)
[6:14:54 AM]  info: - test_example_true_true_30_0 (text_logging_test.TestTextLoggingExample.test_example_true_true_30_0)
[6:14:54 AM]  info: - test_example_true_true_50_0 (text_logging_test.TestTextLoggingExample.test_example_true_true_50_0)
[6:14:54 AM]  
[6:14:54 AM]  Running tests...
[6:14:54 AM]  ----------------------------------------------------------------------
[6:14:54 AM]  ..WARNING:drake:Test Warn message
[6:14:54 AM]  ERROR:drake:Test Error message
[6:14:54 AM]  CRITICAL:drake:Test Critical message
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 393, in _maybe_collect_coverage
[6:14:54 AM]      yield
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 534, in main
[6:14:54 AM]      sys.exit(0)
[6:14:54 AM]  SystemExit: 0
[6:14:54 AM]  
[6:14:54 AM]  During handling of the above exception, another exception occurred:
[6:14:54 AM]  
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 114, in _execute
[6:14:54 AM]      return self.con.execute(sql, parameters)    # type: ignore[arg-type]
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]  sqlite3.OperationalError: no such table: coverage_schema
[6:14:54 AM]  
[6:14:54 AM]  During handling of the above exception, another exception occurred:
[6:14:54 AM]  
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 119, in _execute
[6:14:54 AM]      return self.con.execute(sql, parameters)    # type: ignore[arg-type]
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]  sqlite3.OperationalError: no such table: coverage_schema
[6:14:54 AM]  
[6:14:54 AM]  The above exception was the direct cause of the following exception:
[6:14:54 AM]  
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqldata.py", line 293, in _read_db
[6:14:54 AM]      row = db.execute_one("select version from coverage_schema")
[6:14:54 AM]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 186, in execute_one
[6:14:54 AM]      with self.execute(sql, parameters) as cur:
[6:14:54 AM]    File "/usr/lib/python3.12/contextlib.py", line 137, in __enter__
[6:14:54 AM]      return next(self.gen)
[6:14:54 AM]             ^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 149, in execute
[6:14:54 AM]      cur = self._execute(sql, parameters)
[6:14:54 AM]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 137, in _execute
[6:14:54 AM]      raise DataError(f"Couldn't use data file {self.filename!r}: {msg}") from exc
[6:14:54 AM]  coverage.exceptions.DataError: Couldn't use data file '/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/.coverage': no such table: coverage_schema
[6:14:54 AM]  
[6:14:54 AM]  During handling of the above exception, another exception occurred:
[6:14:54 AM]  
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 537, in <module>
[6:14:54 AM]      main()
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 528, in main
[6:14:54 AM]      with _maybe_collect_coverage(enable=coverage_enabled):
[6:14:54 AM]    File "/usr/lib/python3.12/contextlib.py", line 158, in __exit__
[6:14:54 AM]      self.gen.throw(value)
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 398, in _maybe_collect_coverage
[6:14:54 AM]      cov.lcov_report(
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/control.py", line 1277, in lcov_report
[6:14:54 AM]      return render_report(self.config.lcov_output, LcovReporter(self), morfs, self._message)
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/report_core.py", line 61, in render_report
[6:14:54 AM]      ret = reporter.report(morfs, outfile=outfile)
[6:14:54 AM]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/lcovreport.py", line 46, in report
[6:14:54 AM]      self.coverage.get_data()
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/control.py", line 864, in get_data
[6:14:54 AM]      if self._collector.flush_data():
[6:14:54 AM]         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/collector.py", line 532, in flush_data
[6:14:54 AM]      self.covdata.add_arcs(self.mapped_file_dict(arc_data))
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqldata.py", line 124, in _wrapped
[6:14:54 AM]      return method(self, *args, **kwargs)
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqldata.py", line 533, in add_arcs
[6:14:54 AM]      self._choose_lines_or_arcs(arcs=True)
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqldata.py", line 564, in _choose_lines_or_arcs
[6:14:54 AM]      with self._connect() as con:
[6:14:54 AM]           ^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqldata.py", line 344, in _connect
[6:14:54 AM]      self._open_db()
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqldata.py", line 287, in _open_db
[6:14:54 AM]      self._read_db()
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqldata.py", line 297, in _read_db
[6:14:54 AM]      self._init_db(db)
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqldata.py", line 326, in _init_db
[6:14:54 AM]      db.executescript(SCHEMA)
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 225, in executescript
[6:14:54 AM]      self.con.executescript(script).close()
[6:14:54 AM]      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]  sqlite3.OperationalError: table coverage_schema already exists
[6:14:54 AM]  
[6:14:54 AM]  E.F.F[2026-04-22 10:14:40.230] [console] [info] Test Info message
[6:14:54 AM]  [2026-04-22 10:14:40.230] [console] [warning] Test Warn message
[6:14:54 AM]  [2026-04-22 10:14:40.230] [console] [error] Test Error message
[6:14:54 AM]  [2026-04-22 10:14:40.230] [console] [critical] Test Critical message
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 393, in _maybe_collect_coverage
[6:14:54 AM]      yield
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 534, in main
[6:14:54 AM]      sys.exit(0)
[6:14:54 AM]  SystemExit: 0
[6:14:54 AM]  
[6:14:54 AM]  During handling of the above exception, another exception occurred:
[6:14:54 AM]  
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 114, in _execute
[6:14:54 AM]      return self.con.execute(sql, parameters)    # type: ignore[arg-type]
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]  sqlite3.OperationalError: no such table: arc
[6:14:54 AM]  
[6:14:54 AM]  During handling of the above exception, another exception occurred:
[6:14:54 AM]  
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 119, in _execute
[6:14:54 AM]      return self.con.execute(sql, parameters)    # type: ignore[arg-type]
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]  sqlite3.OperationalError: no such table: arc
[6:14:54 AM]  
[6:14:54 AM]  The above exception was the direct cause of the following exception:
[6:14:54 AM]  
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 537, in <module>
[6:14:54 AM]      main()
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 528, in main
[6:14:54 AM]      with _maybe_collect_coverage(enable=coverage_enabled):
[6:14:54 AM]    File "/usr/lib/python3.12/contextlib.py", line 158, in __exit__
[6:14:54 AM]      self.gen.throw(value)
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 398, in _maybe_collect_coverage
[6:14:54 AM]      cov.lcov_report(
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/control.py", line 1277, in lcov_report
[6:14:54 AM]      return render_report(self.config.lcov_output, LcovReporter(self), morfs, self._message)
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/report_core.py", line 61, in render_report
[6:14:54 AM]      ret = reporter.report(morfs, outfile=outfile)
[6:14:54 AM]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/lcovreport.py", line 50, in report
[6:14:54 AM]      self.get_lcov(fr, analysis, outfile)
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/lcovreport.py", line 124, in get_lcov
[6:14:54 AM]      branch_stats = analysis.branch_stats()
[6:14:54 AM]                     ^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/results.py", line 172, in branch_stats
[6:14:54 AM]      missing_arcs = self.missing_branch_arcs()
[6:14:54 AM]                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/results.py", line 143, in missing_branch_arcs
[6:14:54 AM]      missing = self.arcs_missing()
[6:14:54 AM]                ^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/results.py", line 103, in arcs_missing
[6:14:54 AM]      executed = self.arcs_executed()
[6:14:54 AM]                 ^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/results.py", line 96, in arcs_executed
[6:14:54 AM]      executed = self.data.arcs(self.filename) or []
[6:14:54 AM]                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqldata.py", line 1006, in arcs
[6:14:54 AM]      with con.execute(query, data) as cur:
[6:14:54 AM]    File "/usr/lib/python3.12/contextlib.py", line 137, in __enter__
[6:14:54 AM]      return next(self.gen)
[6:14:54 AM]             ^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 149, in execute
[6:14:54 AM]      cur = self._execute(sql, parameters)
[6:14:54 AM]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 137, in _execute
[6:14:54 AM]      raise DataError(f"Couldn't use data file {self.filename!r}: {msg}") from exc
[6:14:54 AM]  coverage.exceptions.DataError: Couldn't use data file '/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/.coverage': no such table: arc
[6:14:54 AM]  
[6:14:54 AM]  E[2026-04-22 10:14:41,540] [drake] [TRACE] Test Trace message
[6:14:54 AM]  [2026-04-22 10:14:41,541] [drake] [DEBUG] Test Debug message
[6:14:54 AM]  [2026-04-22 10:14:41,541] [drake] [INFO] Test Info message
[6:14:54 AM]  [2026-04-22 10:14:41,541] [drake] [WARNING] Test Warn message
[6:14:54 AM]  [2026-04-22 10:14:41,541] [drake] [ERROR] Test Error message
[6:14:54 AM]  [2026-04-22 10:14:41,541] [drake] [CRITICAL] Test Critical message
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 393, in _maybe_collect_coverage
[6:14:54 AM]      yield
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 534, in main
[6:14:54 AM]      sys.exit(0)
[6:14:54 AM]  SystemExit: 0
[6:14:54 AM]  
[6:14:54 AM]  During handling of the above exception, another exception occurred:
[6:14:54 AM]  
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 114, in _execute
[6:14:54 AM]      return self.con.execute(sql, parameters)    # type: ignore[arg-type]
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]  sqlite3.IntegrityError: UNIQUE constraint failed: context.context
[6:14:54 AM]  
[6:14:54 AM]  During handling of the above exception, another exception occurred:
[6:14:54 AM]  
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 119, in _execute
[6:14:54 AM]      return self.con.execute(sql, parameters)    # type: ignore[arg-type]
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]  sqlite3.IntegrityError: UNIQUE constraint failed: context.context
[6:14:54 AM]  
[6:14:54 AM]  The above exception was the direct cause of the following exception:
[6:14:54 AM]  
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 537, in <module>
[6:14:54 AM]      main()
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 528, in main
[6:14:54 AM]      with _maybe_collect_coverage(enable=coverage_enabled):
[6:14:54 AM]    File "/usr/lib/python3.12/contextlib.py", line 158, in __exit__
[6:14:54 AM]      self.gen.throw(value)
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 398, in _maybe_collect_coverage
[6:14:54 AM]      cov.lcov_report(
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/control.py", line 1277, in lcov_report
[6:14:54 AM]      return render_report(self.config.lcov_output, LcovReporter(self), morfs, self._message)
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/report_core.py", line 61, in render_report
[6:14:54 AM]      ret = reporter.report(morfs, outfile=outfile)
[6:14:54 AM]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/lcovreport.py", line 46, in report
[6:14:54 AM]      self.coverage.get_data()
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/control.py", line 864, in get_data
[6:14:54 AM]      if self._collector.flush_data():
[6:14:54 AM]         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/collector.py", line 532, in flush_data
[6:14:54 AM]      self.covdata.add_arcs(self.mapped_file_dict(arc_data))
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqldata.py", line 124, in _wrapped
[6:14:54 AM]      return method(self, *args, **kwargs)
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqldata.py", line 537, in add_arcs
[6:14:54 AM]      self._set_context_id()
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqldata.py", line 457, in _set_context_id
[6:14:54 AM]      self._current_context_id = con.execute_for_rowid(
[6:14:54 AM]                                 ^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 170, in execute_for_rowid
[6:14:54 AM]      with self.execute(sql, parameters) as cur:
[6:14:54 AM]    File "/usr/lib/python3.12/contextlib.py", line 137, in __enter__
[6:14:54 AM]      return next(self.gen)
[6:14:54 AM]             ^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 149, in execute
[6:14:54 AM]      cur = self._execute(sql, parameters)
[6:14:54 AM]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 137, in _execute
[6:14:54 AM]      raise DataError(f"Couldn't use data file {self.filename!r}: {msg}") from exc
[6:14:54 AM]  coverage.exceptions.DataError: Couldn't use data file '/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/.coverage': UNIQUE constraint failed: context.context
[6:14:54 AM]  
[6:14:54 AM]  E[2026-04-22 10:14:42,212] [drake] [INFO] Test Info message
[6:14:54 AM]  [2026-04-22 10:14:42,212] [drake] [WARNING] Test Warn message
[6:14:54 AM]  [2026-04-22 10:14:42,212] [drake] [ERROR] Test Error message
[6:14:54 AM]  [2026-04-22 10:14:42,212] [drake] [CRITICAL] Test Critical message
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 393, in _maybe_collect_coverage
[6:14:54 AM]      yield
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 534, in main
[6:14:54 AM]      sys.exit(0)
[6:14:54 AM]  SystemExit: 0
[6:14:54 AM]  
[6:14:54 AM]  During handling of the above exception, another exception occurred:
[6:14:54 AM]  
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 114, in _execute
[6:14:54 AM]      return self.con.execute(sql, parameters)    # type: ignore[arg-type]
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]  sqlite3.OperationalError: no such table: arc
[6:14:54 AM]  
[6:14:54 AM]  During handling of the above exception, another exception occurred:
[6:14:54 AM]  
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 119, in _execute
[6:14:54 AM]      return self.con.execute(sql, parameters)    # type: ignore[arg-type]
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]  sqlite3.OperationalError: no such table: arc
[6:14:54 AM]  
[6:14:54 AM]  The above exception was the direct cause of the following exception:
[6:14:54 AM]  
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 537, in <module>
[6:14:54 AM]      main()
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 528, in main
[6:14:54 AM]      with _maybe_collect_coverage(enable=coverage_enabled):
[6:14:54 AM]    File "/usr/lib/python3.12/contextlib.py", line 158, in __exit__
[6:14:54 AM]      self.gen.throw(value)
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 398, in _maybe_collect_coverage
[6:14:54 AM]      cov.lcov_report(
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/control.py", line 1277, in lcov_report
[6:14:54 AM]      return render_report(self.config.lcov_output, LcovReporter(self), morfs, self._message)
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/report_core.py", line 61, in render_report
[6:14:54 AM]      ret = reporter.report(morfs, outfile=outfile)
[6:14:54 AM]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/lcovreport.py", line 50, in report
[6:14:54 AM]      self.get_lcov(fr, analysis, outfile)
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/lcovreport.py", line 124, in get_lcov
[6:14:54 AM]      branch_stats = analysis.branch_stats()
[6:14:54 AM]                     ^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/results.py", line 172, in branch_stats
[6:14:54 AM]      missing_arcs = self.missing_branch_arcs()
[6:14:54 AM]                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/results.py", line 143, in missing_branch_arcs
[6:14:54 AM]      missing = self.arcs_missing()
[6:14:54 AM]                ^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/results.py", line 103, in arcs_missing
[6:14:54 AM]      executed = self.arcs_executed()
[6:14:54 AM]                 ^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/results.py", line 96, in arcs_executed
[6:14:54 AM]      executed = self.data.arcs(self.filename) or []
[6:14:54 AM]                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqldata.py", line 1006, in arcs
[6:14:54 AM]      with con.execute(query, data) as cur:
[6:14:54 AM]    File "/usr/lib/python3.12/contextlib.py", line 137, in __enter__
[6:14:54 AM]      return next(self.gen)
[6:14:54 AM]             ^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 149, in execute
[6:14:54 AM]      cur = self._execute(sql, parameters)
[6:14:54 AM]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 137, in _execute
[6:14:54 AM]      raise DataError(f"Couldn't use data file {self.filename!r}: {msg}") from exc
[6:14:54 AM]  coverage.exceptions.DataError: Couldn't use data file '/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/.coverage': no such table: arc
[6:14:54 AM]  
[6:14:54 AM]  E[2026-04-22 10:14:43,976] [drake] [WARNING] Test Warn message
[6:14:54 AM]  [2026-04-22 10:14:43,976] [drake] [ERROR] Test Error message
[6:14:54 AM]  [2026-04-22 10:14:43,976] [drake] [CRITICAL] Test Critical message
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 393, in _maybe_collect_coverage
[6:14:54 AM]      yield
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 534, in main
[6:14:54 AM]      sys.exit(0)
[6:14:54 AM]  SystemExit: 0
[6:14:54 AM]  
[6:14:54 AM]  During handling of the above exception, another exception occurred:
[6:14:54 AM]  
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 114, in _execute
[6:14:54 AM]      return self.con.execute(sql, parameters)    # type: ignore[arg-type]
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]  sqlite3.IntegrityError: UNIQUE constraint failed: context.context
[6:14:54 AM]  
[6:14:54 AM]  During handling of the above exception, another exception occurred:
[6:14:54 AM]  
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 119, in _execute
[6:14:54 AM]      return self.con.execute(sql, parameters)    # type: ignore[arg-type]
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]  sqlite3.IntegrityError: UNIQUE constraint failed: context.context
[6:14:54 AM]  
[6:14:54 AM]  The above exception was the direct cause of the following exception:
[6:14:54 AM]  
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 537, in <module>
[6:14:54 AM]      main()
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 528, in main
[6:14:54 AM]      with _maybe_collect_coverage(enable=coverage_enabled):
[6:14:54 AM]    File "/usr/lib/python3.12/contextlib.py", line 158, in __exit__
[6:14:54 AM]      self.gen.throw(value)
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 398, in _maybe_collect_coverage
[6:14:54 AM]      cov.lcov_report(
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/control.py", line 1277, in lcov_report
[6:14:54 AM]      return render_report(self.config.lcov_output, LcovReporter(self), morfs, self._message)
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/report_core.py", line 61, in render_report
[6:14:54 AM]      ret = reporter.report(morfs, outfile=outfile)
[6:14:54 AM]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/lcovreport.py", line 46, in report
[6:14:54 AM]      self.coverage.get_data()
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/control.py", line 864, in get_data
[6:14:54 AM]      if self._collector.flush_data():
[6:14:54 AM]         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/collector.py", line 532, in flush_data
[6:14:54 AM]      self.covdata.add_arcs(self.mapped_file_dict(arc_data))
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqldata.py", line 124, in _wrapped
[6:14:54 AM]      return method(self, *args, **kwargs)
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqldata.py", line 537, in add_arcs
[6:14:54 AM]      self._set_context_id()
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqldata.py", line 457, in _set_context_id
[6:14:54 AM]      self._current_context_id = con.execute_for_rowid(
[6:14:54 AM]                                 ^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 170, in execute_for_rowid
[6:14:54 AM]      with self.execute(sql, parameters) as cur:
[6:14:54 AM]    File "/usr/lib/python3.12/contextlib.py", line 137, in __enter__
[6:14:54 AM]      return next(self.gen)
[6:14:54 AM]             ^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 149, in execute
[6:14:54 AM]      cur = self._execute(sql, parameters)
[6:14:54 AM]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 137, in _execute
[6:14:54 AM]      raise DataError(f"Couldn't use data file {self.filename!r}: {msg}") from exc
[6:14:54 AM]  coverage.exceptions.DataError: Couldn't use data file '/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/.coverage': UNIQUE constraint failed: context.context
[6:14:54 AM]  
[6:14:54 AM]  E[2026-04-22 10:14:44,643] [drake] [CRITICAL] Test Critical message
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 393, in _maybe_collect_coverage
[6:14:54 AM]      yield
[6:14:54 AM]    File "/media/ephemeral0/ubuntu/workspace/linux-noble-gcc-bazel-nightly-coverage/_bazel_ubuntu/fd438e4294a5517b3824228c83cfb7fc/execroot/_main/bazel-out/k8-dbg/bin/bindings/pydrake/common/py/text_logging_test.runfiles/_main/bindings/pydrake/common/_text_logging_example_stage2_bootstrap.py", line 534, in main
[6:14:54 AM]      sys.exit(0)
[6:14:54 AM]  SystemExit: 0
[6:14:54 AM]  
[6:14:54 AM]  During handling of the above exception, another exception occurred:
[6:14:54 AM]  
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 114, in _execute
[6:14:54 AM]      return self.con.execute(sql, parameters)    # type: ignore[arg-type]
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]  sqlite3.OperationalError: no such table: tracer
[6:14:54 AM]  
[6:14:54 AM]  During handling of the above exception, another exception occurred:
[6:14:54 AM]  
[6:14:54 AM]  Traceback (most recent call last):
[6:14:54 AM]    File "/usr/lib/python3/dist-packages/coverage/sqlitedb.py", line 119, in _execute
[6:14:54 AM]      return self.con.execute(sql, parameters)    # type: ignore[arg-type]
[6:14:54 AM]             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[6:14:54 AM]  sqlite3.OperationalError: no such table: tracer
```

--- a/tools/skylark/drake_py.bzl
+++ b/tools/skylark/drake_py.bzl
@@ -270,7 +270,7 @@ def drake_py_test(
         srcs = ["test/%s.py" % name]
     if kwargs.get("shard_count") != None:
         fail("Only drake_py_unittest can use sharding")
-    shard_count = kwargs.pop("_drake_py_unittest_shard_count", None)
+    kwargs["shard_count"] = kwargs.pop("_drake_py_unittest_shard_count", None)
 
     kwargs = incorporate_allow_network(kwargs, allow_network = allow_network)
     kwargs = incorporate_display(kwargs, display = display)
@@ -293,7 +293,6 @@ def drake_py_test(
         py_target = py_test,
         isolate = isolate,
         size = size,
-        shard_count = shard_count,
         srcs = srcs,
         deps = deps,
         target_compatible_with = target_compatible_with,

--- a/tools/skylark/kwargs.bzl
+++ b/tools/skylark/kwargs.bzl
@@ -121,9 +121,20 @@ def incorporate_allow_network(kwargs, *, allow_network):
     return kwargs
 
 def incorporate_test_weight_heuristics(kwargs):
-    # kcov is only appropriate for small-sized unit tests. If a test needs a
-    # shard_count or a special timeout, we assume it is not small.
-    if "shard_count" in kwargs or "timeout" in kwargs:
+    # If a test needs a shard_count or a special timeout, we assume it is not
+    # small -- kcov is only appropriate for small-sized unit tests.
+    required = {
+        "shard_count": 1,
+        "size": "small",
+        "timeout": "short",
+    }
+    big = False
+    for arg_name, required_value in required.items():
+        value = kwargs.get(arg_name, None) or required_value
+        if value != required_value:
+            big = True
+            break
+    if big:
         kwargs = amend(kwargs, "opt_out_conditions", append = [
             "//tools/kcov:enabled",
         ])

